crates/turbopack-core/src/chunk/availability_info.rs

```diff
@@ -1,19 +1,56 @@
+use std::ops::{Add, AddAssign};
+
 use turbo_tasks::Vc;
 
 use super::available_modules::AvailableAssets;
 use crate::module::Module;
 
+#[turbo_tasks::value(shared)]
+#[derive(Copy, Clone, Debug)]
+pub enum AvailabilityInfoNeeds {
+    None,
+    Root,
+    AvailableModules,
+    Complete,
+}
+
+impl Add for AvailabilityInfoNeeds {
+    type Output = AvailabilityInfoNeeds;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        match (self, rhs) {
+            (Self::None, rhs) => rhs,
+            (lhs, Self::None) => lhs,
+            (Self::Complete, _) | (_, Self::Complete) => Self::Complete,
+            (Self::Root, Self::Root) => Self::Root,
+            (Self::AvailableModules, Self::AvailableModules) => Self::AvailableModules,
+            (Self::Root, Self::AvailableModules) | (Self::AvailableModules, Self::Root) => {
+                Self::Complete
+            }
+        }
+    }
+}
+
+impl AddAssign for AvailabilityInfoNeeds {
+    fn add_assign(&mut self, rhs: Self) {
+        *self = *self + rhs;
+    }
+}
+
 #[turbo_tasks::value(serialization = "auto_for_input")]
 #[derive(PartialOrd, Ord, Hash, Clone, Copy, Debug)]
 pub enum AvailabilityInfo {
```

Comment:
Docs. What is the difference between `Complete` and `OnlyAvailableModules`?

Reply:
`Complete` also contains the root module.